src/mono/mono/mini/interp/interp.c

```diff
@@ -5358,6 +5358,16 @@ interp_exec_method (InterpFrame *frame, ThreadContext *context, FrameClauseArgs
 			ip++;
 			MINT_IN_BREAK;
 		}
+		MINT_IN_CASE(MINT_INTRINS_BITOPS_LOG2_32) {
+			sp [-1].data.i = 31 ^ (gint32)__builtin_clz ((guint32)sp [-1].data.i | 1);
```

Comment:
`__builtin_clz` uses `bsr` since we don't set `-march=native` or `-mattr+` https://godbolt.org/z/bMqcM5
`lzcnt` makes it even faster.

Reply:
This might need an ifdef gcc or something.

Reply:
@vargaz works fine on clang and should be correctly expanded for Arm

Reply:
ah, forgot about MSVC

Reply:
With MSVC, we can use _BitScan{Forward,Reverse} with [this conversion](https://stackoverflow.com/a/20468180/863980). (can be wrapped in a macro?)

Reply:
@EgorBo you could extract `sgen_clz` out of sgen into utils and reuse it here. We have the logic there to use _BitScan on msvc